### ISSUE TITLE
aysnc compiled code has unnecessary code

### ISSUE THREAD

**Issue by welefen** [babel/babel#2049]
source code:
```
var fn = async function(){
  var data = await Promise.resolve(1);
}
```

compiled code:
```
"use strict";

var _regeneratorRuntime = require("babel-runtime/regenerator")["default"];

var _Promise = require("babel-runtime/core-js/promise")["default"];

var fn = function fn() {
  var data;
  return _regeneratorRuntime.async(function fn$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
        context$1$0.next = 2;
        return _regeneratorRuntime.awrap(_Promise.resolve(1));

      case 2:
        data = context$1$0.sent;

      case 3:
      case "end":
        return context$1$0.stop();
    }
  }, null, this);
};

```

code `case 3:` is unnecessary.

**Comment by fkling** [comment]
Babel uses [regenerator](https://github.com/facebook/regenerator) to compile async functions. If any, you want to report it there.

**Comment by welefen** [comment]
I have report a issue to regenerator.  https://github.com/facebook/regenerator/issues/211

**Comment by sebmck** [comment]
Closed via facebook/regenerator#211.
